src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.cs

```diff
@@ -20,5 +45,173 @@ internal static partial class Crypto
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_UpRefEvpPkey")]
         internal static extern int UpRefEvpPkey(SafeEvpPKeyHandle handle);
+
+        [DllImport(Libraries.CryptoNative)]
+        private static extern unsafe SafeEvpPKeyHandle CryptoNative_DecodeSubjectPublicKeyInfo(
+            byte* buf,
+            int len,
+            int algId);
+
+        [DllImport(Libraries.CryptoNative)]
+        private static extern unsafe SafeEvpPKeyHandle CryptoNative_DecodePkcs8PrivateKey(
+            byte* buf,
+            int len,
+            int algId);
+
+        internal static unsafe SafeEvpPKeyHandle DecodeSubjectPublicKeyInfo(
+            ReadOnlySpan<byte> source,
```

Comment:
Nit: Since `CryptoNative_DecodeSubjectPublicKeyInfo` requires _buf != nullptr_ and _len \> 0_, may want a `Debug.Assert(!source.IsEmpty);` statement here.

(Same comment for related methods in this file.)